libraries/csharp-net22-runtime/context/Pipe/Program.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+namespace Pipe
+{
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            if (args.Length == 0)
+            {
+                throw new Exception(
+                    "no algorithm directory argument found. Please provide the path to a valid C# algorithm.");
+            }
+            string sysPath = args[0];
+            Config config = new Config(sysPath);
+            Module algoModule;
+            try
+            {
+                algoModule = new Module(config);
+            }
+            catch (Exception e)
+            {
+                ExceptionResponse response = new ExceptionResponse(e);
+                Write.WriteJsonToPipe(response);
+                Console.Out.WriteLine("PIPE_INIT_FAILED");
+                return -1;
+            }
+            Console.Out.WriteLine("PIPE_INIT_COMPLETE");
+            
+            string readLine;
+            while ((readLine = Console.In.ReadLine()) != null && readLine != "")
+            {
+                Request request = new Request(readLine);
+                object response = null;
+                try
+                {
+                    object result = algoModule.AttemptExecute(request);
+                     response = new Response(result, "json");
+                }
+                catch (Exception e)

```

Comment:
i'm not sure if there's a "non-fatal" exception we should be catching here (like how catching throwable in java is bad)

Reply:
oh yea good point, I'll investigate

Reply:
https://stackoverflow.com/questions/26423380/intentionally-cause-a-fatal-exception
https://docs.microsoft.com/en-us/dotnet/api/system.environment.failfast - this is the way of terminating without throwing exceptions. It seems like `Exception` is a sufficient catchall type.
We can certainly revisit this later in the algorithmia-c-sharp-dev package.